Review (#19)
herflis/concepts-temp: content/review/content-tree.md

---
title: "Content Tree"
metaTitle: "sensenet - Content Tree"
metaDescription: "sensenet Content Tree"
---

> sensenet content repository is basically a tree structure of the various stored content. A specific content is identified by a unique number (id) and also by its path. The root of the tree is at `/Root` path, all other content is placed somewhere under this root content - for example the users are placed at `/Root/IMS`.

Author: ```suggestion
sensenet content repository is basically a tree structure of the various stored content. A specific content is identified by a unique number (id) and also by its path (which is determined by its place in the tree). The root of the tree is at the `/Root` path, all other contents are placed somewhere under this root content - for example the users are placed at `/Root/IMS`.
```

# Structured storage of content

# Content access and url resolution

> Every content in the content repository is identified by its unique Id and its path. You cannot change a content Id, but you can move a content to another folder and thus change its path. The tree structure of the repository makes it possible to use the path as a link to the content.

Reviewer: ```suggestion
The tree structure of the repository makes it possible to use the path as a link to the content.
```

should be extended with more details

# Advantages of the tree structure

> - inheritance (type, permission)

Author: I think these stuff would be more useful if we would put them in whole sentences

---
title: "Content Tree"
metaTitle: "sensenet - Content Tree"
metaDescription: "sensenet Content Tree"
---

sensenet content repository is basically a tree structure of the various stored content (e.g. folders, documents, user, tasks, etc.). A specific content is identified by a unique number (id) and also by its path. The root of the tree is at `/Root` path, all other content is placed somewhere under this root content - for example the users are placed at `/Root/IMS`.

# Structured storage of content

> Tree structure lets you orginize, group your content easily by creating separate subtrees. It's not only important because of the possibility the configure content accessibility for different types of users on different parts of the tree, but also because of the inheritence of several things like content types' underlying business logic, metadata field configurations and settings.

Reviewer: ```suggestion
Tree structure lets you organize, group your content easily by creating separate subtrees. It's not only important because of the possibility to configure content accessibility for different types of users on different parts of the tree, but also because of the inheritence of several things like content types' underlying business logic, metadata field configurations and settings.
```

This kind of data storage makes it possible to store content of multiple different projects (public website, intranet, document management app, etc.) in one repository. By creating multiple subtrees with folders or workspaces you can build up custom content structures and every subtree can have its own permission settings, roles and configurations, considering who can access it and what it will be used for.

# Searching in subtrees

> sensenet's search engine provides you an easy and perfomant way to query content in the repository. You can make it even more efficient building seperate subtrees for separate use cases. This way the system will only search those part of the tree that are related making the search itself much faster and efficient.

Reviewer: ```suggestion
sensenet's search engine provides you an easy and perfomant way to query content in the repository. You can make it even more efficient building separate subtrees for separate use cases. This way the system will only search those part of the tree that are related making the search itself much faster and efficient.
```

# Content access and url resolution

As it is mentioned above every content in the content repository is not only identified by its unique Id but also its path. If you move a content to another folder thus change its path, so the tree structure of the repository makes it possible to use the path as a link to the content.

# Permission inheritance

> As in sensenet content is stored in a huge tree with a single root, when you set a permission on a container item on a higher level (e.g. a document library or a workspace) it will be inherited by its children. This way you do not have to set permissions on every subfolder or document, because all content inherit permissions from their parent.

Reviewer: ```suggestion
As in sensenet content is stored in a huge tree with a single root, when you set a permission on a container item on a higher level (e.g. a document library or a workspace) it will be inherited by its children. This way you do not have to set permissions on every subfolder or document, because all content inherits permissions from their parent.
```

Inherited permissions cannot be changed without breaking the inheritance, but you can set additional permissions for the same identity (e.g. grant `Open` permission to someone who already has `See` permissions inherited).

Sometimes you may need to set local only permissions that are not propagated to children and it is also possible to do with sensenet, check the other docs about the permission system.

# Settings inheritance

> Settings are created for administrators or editors to let them customize the behavior of a certain feature. In sensenet these settings are stored as content in the content cepository so you can take all the advantages of the tree sturcture in case of settings as well.

Reviewer: ```suggestion
Settings are created for administrators or editors to let them customize the behavior of a certain feature. In sensenet these settings are stored as content in the content repository so you can take all the advantages of the tree sturcture in case of settings as well.
```
